yarn-project/simulator/src/avm/opcodes/external_calls.test.ts

```diff
@@ -58,6 +58,42 @@ describe('External Calls', () => {
       expect(inst.serialize()).toEqual(buf);
     });
 
+    it('Call to non-existent bytecode returns failure', async () => {
+      const gasOffset = 0;
+      const l2Gas = 2e6;
+      const daGas = 3e6;
+      const addrOffset = 2;
+      const addr = new Fr(123456n);
+      const argsOffset = 3;
+      const args = [new Field(1), new Field(2), new Field(3)];
+      const argsSize = args.length;
+      const argsSizeOffset = 20;
+      const successOffset = 6;
+
+      const { l2GasLeft: initialL2Gas, daGasLeft: initialDaGas } = context.machineState;
+
+      context.machineState.memory.set(0, new Field(l2Gas));
+      context.machineState.memory.set(1, new Field(daGas));
+      context.machineState.memory.set(2, new Field(addr));
+      context.machineState.memory.set(argsSizeOffset, new Uint32(argsSize));
+      context.machineState.memory.setSlice(3, args);
+
+      const instruction = new Call(/*indirect=*/ 0, gasOffset, addrOffset, argsOffset, argsSizeOffset, successOffset);
+      await instruction.execute(context);
+
+      const successValue = context.machineState.memory.get(successOffset);
+      expect(successValue).toEqual(new Uint1(0n)); // failure, contract non-existent!
+
+      const retValue = context.machineState.nestedReturndata;
+      expect(retValue).toEqual([]);
+
+      // should only charge gas for the call instruction itself
+      // (all gas allocated to the nested call should be refunded)
+      const callGasCost = instruction.gasCost(argsSize);
+      expect(context.machineState.l2GasLeft).toEqual(initialL2Gas - callGasCost.l2Gas);
+      expect(context.machineState.daGasLeft).toEqual(initialDaGas - callGasCost.daGas);
```

Comment:
changed `instruction.gasCost` from protected to public so that I can confirm here that the external call is charging no more than the cost of the call opcode itself. Is that an acceptable decision?

Reply:
undid